ganga/GangaCore/GPIDev/Lib/Job/Job.py

```diff
@@ -1457,6 +1477,111 @@ def _doSplitting(self):
+        appmasterconfig = self._getMasterAppConfig()
+
+        logger.info("Re-splitting Job: %s" % fqid)
+
+        self.splitter = new_splitter
+
+        rjobs = self.splitter.validatedSplit(self)
+        if not rjobs:
+            raise SplitterError("Splitter '%s' failed to produce any subjobs from re-splitting. Aborting submit" % (getName(self.splitter),))
+
+        index = len(mJob.subjobs)
+
+        logger.debug('First index of new jobs: %s' % index)
+        if rjobs:
+            if not isType(mJob.subjobs, (list, GangaList)):
+                    mJob.subjobs = GangaList()
+            i = index
+            for sj in rjobs:
+                sj.info.uuid = str(uuid.uuid4())
+                sj.status = 'new'
+                sj.time.timenow('new')
+                sj.id = i
+                i += 1
+                mJob.subjobs.append(sj)
+
+            cfg = GangaCore.Utility.Config.getConfig('Configuration')
+            for j in rjobs:
+                if cfg['autoGenerateJobWorkspace']:
+                    j._init_workspace()
+
+            logger.info('submitting %s subjobs', len(rjobs))
+        
+            # Output Files
+            # validate the output files
+            for this_job in rjobs:
+                (validOutputFiles, errorMsg) = this_job.validateOutputfilesOnSubmit()
+                if not validOutputFiles:
+                    raise JobError(errorMsg)
+
+            # configure the application of each subjob
+            appsubconfig = self._getAppSubConfig(rjobs)
+            if appsubconfig is not None:
+                logger.debug("# appsubconfig: %s" % len(appsubconfig))
+
+            # Job Configuration
+            logger.debug("Job Configuration, Job %s:" % self.getFQID('.'))
+
+            # prepare the master job with the correct runtime handler
+            # Calls rtHandler.master_prepare if it hasn't already been called by the master job or by self
+            # Only stored as transient if the master_prepare successfully
+            # completes
+            jobmasterconfig = self._getJobMasterConfig()
+
+            # prepare the subjobs with the runtime handler
+            # Calls the rtHandler.prepare if it hasn't already been called by the master job or by self
+            # Only stored as a transient if the prepare successfully completes
+            jobsubconfig = self._getJobSubConfig(rjobs)
+            logger.debug("# jobsubconfig: %s" % len(jobsubconfig))
+
+            # Submission
+            logger.debug("Submitting to a backend, Job %s:" % self.getFQID('.'))
+
+            # notify monitoring-services
+            self.monitorPrepare_hook(jobsubconfig)
+
+            # submit the job
+            # master_submit has been written as the interface which ganga
+            # should call, not submit directly
+
+            r = self.backend.master_submit( rjobs, jobsubconfig, jobmasterconfig)
+
+            if not r:
+                raise JobManagerError('error during submit')
```

Comment:
Should we remove the newly created subjobs again if there is an error in the submission?

Reply:
probably yes, as long as we have the original they were split from